pkg/streamingpromql/operators/aggregations/quantile.go

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/prometheus/prometheus/blob/main/promql/engine.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Prometheus Authors
+
+package aggregations
+
+import (
+	"math"
+	"unsafe"
+
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/promql/parser/posrange"
+	"github.com/prometheus/prometheus/util/annotations"
+
+	"github.com/grafana/mimir/pkg/streamingpromql/limiting"
+	"github.com/grafana/mimir/pkg/streamingpromql/operators/functions"
+	"github.com/grafana/mimir/pkg/streamingpromql/types"
+	"github.com/grafana/mimir/pkg/util/pool"
+)
+
+type QuantileAggregationGroup struct {
+	heaps []heap // A heap per point in time
+}
+
+type heap struct {
+	points []float64 // All of the floats for this group series at a point in time
+}
+
+const maxExpectedQuantileGroups = 64 // There isn't much science to this
+
+var heapPool = types.NewLimitingBucketedPool(
+	pool.NewBucketedPool(maxExpectedQuantileGroups, func(size int) []heap {
+		return make([]heap, 0, size)
+	}),
+	uint64(unsafe.Sizeof(heap{})),
+	true,
```

Comment:
This isn't necessary - there's no need to clear returned slices given we always set `points` to nil on line 99 (assuming we make the suggested fix there).